src/test/dependency-parser.test.ts

```diff
@@ -0,0 +1,358 @@
+      script: {kind: 'name', name: './scr'},
+    },
+  ],
+  [
+    '!./pkg#scr',
+    '_PPPPP_SSS',
+    {
+      inverted: true,
+      package: {kind: 'path', path: './pkg'},
+      script: {kind: 'name', name: 'scr'},
+    },
+  ],
+  [
+    '!scr',
+    '_SSS',
+    {
+      inverted: true,
+      package: {kind: 'this'},
+      script: {kind: 'name', name: 'scr'},
+    },
+  ],
+  [
+    '!!scr',
+    '_SSSS',
+    {
+      inverted: true,
+      package: {kind: 'this'},
+      script: {kind: 'name', name: '!scr'},
+    },
+  ],
+  [
+    String.raw`\!scr`,
+    String.raw`_SSSS`,
+    {
+      package: {kind: 'this'},
+      script: {kind: 'name', name: '!scr'},
+    },
+  ],
+  [
+    './pkg:scr',
+    'PPPPP_SSS',
+    {
+      package: {kind: 'path', path: './pkg'},
+      script: {kind: 'name', name: 'scr'},
+    },
+  ],
+  [
+    './pkg:foo#scr',
+    'PPPPPPPPP_SSS',
+    {
+      package: {
+        kind: 'path',
+        path: './pkg:foo',
+      },
+      script: {kind: 'name', name: 'scr'},
+    },
+  ],
+  [
+    String.raw`./pkg:foo\#scr`,
+    String.raw`PPPPP_SSSSSSSS`,
+    {
+      package: {kind: 'path', path: './pkg'},
+      script: {kind: 'name', name: 'foo#scr'},
+    },
+  ],
+  [
+    String.raw`./pkg:foo\##scr`,
+    String.raw`PPPPPPPPPPP_SSS`,
+    {
+      package: {kind: 'path', path: './pkg:foo#'},
+      script: {kind: 'name', name: 'scr'},
+    },
+  ],
+  [
+    '#foo',
+    '_SSS',
+    {
+      severity: 'error',
+      message: 'Package specifier cannot be empty',
+    },
+  ],
+  [
+    './pkg:<workspaces>',
+    'PPPPP_EEEEEEEEEEEE',
+    {
+      severity: 'error',
+      message: 'Unknown special script "workspaces"',
+    },
+  ],
+  [
+    './pkg#<>',
+    'PPPPP_EE',
+    {
+      severity: 'error',
+      message: 'Unexpected ">". Escape as "\\>" if you meant it literally.',
+    },
+  ],
+  [
+    '<>',
+    'EE',
+    {
+      severity: 'error',
+      message: 'Unexpected ">". Escape as "\\>" if you meant it literally.',
+    },
+  ],
+  [
+    '<workspaces>#scr',
+    'PPPPPPPPPPPP_SSS',
+    {
+      package: {kind: 'workspaces'},
+      script: {kind: 'name', name: 'scr'},
+    },
+  ],
+  [
+    '<dependencies>#scr',
+    'PPPPPPPPPPPPPP_SSS',
+    {
+      package: {kind: 'dependencies'},
+      script: {kind: 'name', name: 'scr'},
+    },
+  ],
+  [
+    '<dependencies>#<this>',
+    'PPPPPPPPPPPPPP_SSSSSS',
+    {
+      package: {kind: 'dependencies'},
+      script: {kind: 'this'},
+    },
+  ],
+  [
+    String.raw`./pkg\*#scr\*`,
+    String.raw`PPPPPPP_SSSSS`,
+    {
+      package: {kind: 'path', path: './pkg*'},
+      script: {kind: 'name', name: 'scr*'},
+    },
+  ],
+  [
+    String.raw`./pkg\{foo,bar}#scr\{foo,bar}`,
+    String.raw`PPPPPPPPPPPPPPP_SSSSSSSSSSSSS`,
+    {
+      package: {
+        kind: 'path',
+        path: './pkg{foo,bar}',
+      },
+      script: {
+        kind: 'name',
+        name: 'scr{foo,bar}',
+      },
+    },
+  ],
```

Comment:
if/when `*` and `{,}` expands are supported, they wouldn't actually require backslash escapes, would they? these tests are simply demonstrating handling of unnecessary escapes?